### PR TITLE
Add support for getting module eeprom

### PR DESCRIPTION
**Pull request by mickep76**
- Added support for getting module eeprom like SFP's, allows for getting Transceiver Vendor, Product, S/N
- Added an example

### CODE REVIEW

**Comment by mickep76** [comment]
Same as "ethtool -m eth0 hex on"

**Comment by mickep76** [comment]
Started adding module info

**Comment by mickep76** [comment]
That finished support for module eeprom and SFF-8709 info.

File: ethtool.go [hunk 1]
```diff
@@ -117,6 +140,42 @@ func (e *Ethtool) BusInfo(intf string) (string, error) {
 	return string(bytes.Trim(info.bus_info[:], "\x00")), nil
 }
 
+func (e *Ethtool) ModuleEeprom(intf string) (string, error) {
```
Comment (safchain) [inline ethtool.go]: Could you add the documentation for each exported function

File: sff_8436.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package ethtool
```
Comment (safchain) [inline sff_8436.go]: could you add the license header for each new file

File: example/main.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+
+	//	"github.com/safchain/ethtool"
```
Comment (safchain) [inline example/main.go]: correct package has to used, probably debug stuff remaining :)

File: ethtool.go [hunk 4]
```diff
@@ -138,6 +197,44 @@ func (e *Ethtool) getDriverInfo(intf string) (ethtoolDrvInfo, error) {
...
+	}
+
+	eeprom := ethtoolEeprom{
+		cmd:    ETHTOOL_GMODULEEEPROM,
+		len:    modInfo.eeprom_len,
+		offset: 0,
+	}
+
+	if modInfo.eeprom_len > EEPROM_LEN {
+		return ethtoolEeprom{}, ethtoolModInfo{}, fmt.Errorf("eeprom size: %d is larger then buffer size: %d", modInfo.eeprom_len, EEPROM_LEN)
```
Comment (safchain) [inline ethtool.go]: nit: s/then/than/

**Comment by mickep76** [comment]
I'm adding additional code for QFP 8436, so better to merge until after that. I'll address your comments as-well.

**Comment by mickep76** [comment]
I will split out the SFF decode part in a separate package otherwise you'll have to deal with license conflict GPL from orig. ethtool vs Apache. Since the code is basically a copy/paste of their code from C to Go. I have to respect their original license. https://github.com/mickep76/sff-eeprom

**Comment by mickep76** [comment]
Fixed the comments in the review, sry for the delay. I implemented a module decoder separately mostly because it needed a diff. license. I ended up completely re-factoring anyway so I can merge it in to this code base if you think that make sense? (https://github.com/mickep76/go-sff)

**Pull request merged**